climakitae/explore/amy.py

```diff
@@ -203,6 +296,574 @@ def _return_diurnal(y: xr.DataArray) -> xr.DataArray:
+    if has_simulation:
+        n_simulations = len(data.simulation)
+        simulations = data.simulation.values
+    else:
+        n_simulations = 1
+        simulations = [None]
+
+    # Get all available time_delta data (all 30 years)
+    hours_per_day = 24
+    hours_per_year = 8760
+    total_hours = len(data.time_delta)
+    n_years = total_hours // hours_per_year
+
+    print(f"      📊 Processing {total_hours:,} hours ({n_years} years) of data")
+    print(f"      🎯 Computing {q*100:.0f}th percentile for each hour of year")
+
+    # Create hour-of-year coordinate for all data (cycling through 1-8760)
+    hour_of_year_all = np.tile(np.arange(1, hours_per_year + 1), n_years)[:total_hours]
+    data = data.assign_coords(hour_of_year=("time_delta", hour_of_year_all))
+
+    warming_levels = data.warming_level.values
+
+    # Create helper function to extract meaningful simulation labels
+    def _get_simulation_label(sim, sim_idx):
+        """Extract meaningful simulation label from simulation identifier."""
+        if sim is None:
+            return f"Sim_{sim_idx+1}"
+
+        sim_str = str(sim)
+        if "WRF_" in sim_str:
+            # Extract the GCM model name (e.g., CESM2, CNRM-ESM2-1, etc.)
+            parts = sim_str.split("_")
+            if len(parts) >= 2:
+                return parts[1]  # Get the GCM name
+            else:
+                return f"Sim_{sim_idx+1}"
+        else:
+            return sim_str.split("_")[0] if "_" in sim_str else sim_str
+
+    # Process all data using quantile computation across years
+    print(
+        f"      ⚙️  Computing quantiles for {len(warming_levels)} warming level(s) and {n_simulations} simulation(s)"
+    )
+
+    # Initialize storage for profiles
+    profile_data = {}
+
+    # Progress tracking
+    total_combinations = len(warming_levels) * n_simulations
+    with tqdm(
+        total=total_combinations,
+        desc="      Computing profiles",
+        unit="combo",
+        leave=False,
+    ) as pbar:
+
+        for wl_idx, wl in enumerate(warming_levels):
+            for sim_idx, sim in enumerate(simulations):
+                # Get simulation label
+                sim_label = _get_simulation_label(sim, sim_idx)
+
+                # Select data for this warming level and simulation combination
+                if has_simulation:
+                    subset_data = data.isel(warming_level=wl_idx, simulation=sim_idx)
+                else:
+                    subset_data = data.isel(warming_level=wl_idx)
+
+                # Group by hour_of_year and find the actual data value closest to the quantile
+                # This gives us the actual data point closest to the q-th quantile for each of the 8760 hours
+                # Load data to avoid dask chunking issues with quantile
+                if hasattr(subset_data.data, "chunks"):
+                    # If it's a dask array, load it into memory
+                    subset_data = subset_data.compute()
```

Comment:
Somewhat useful, somewhat nice-to-have for the future, depending on user's data needs:

Including an interactive progress bar as data is being computed assures users that the data compute is making progress live. This can prevent users from worrying that their code is stuck just because they requested a really large spatial extent or many warming levels.

Using `ProgressBar` from Dask doesn't integrate well with this file's usage of `tqdm`, so something like https://stackoverflow.com/a/70131104 from StackOverflow could be nice to use in place of `.compute()` here.